src/connect_to_peers.rs

```diff
@@ -757,6 +780,80 @@ mod connect_tests {
         Ok(())
     }
 
+    #[traced_test]
+    #[tokio::test]
+    async fn bootstrap_node_refuses_reconnects_within_disconnect_cooldown_period() -> Result<()> {
+        let network = Network::Main;
+        let reconnect_cooldown = Duration::from_secs(3);
```

Comment:
It looks to me like there's a risk that this test becomes flaky, because of your assumptions about how fast it's run, which may not hold up in all scenarios. But we can just fix that if it becomes a problem IMO.

Reply:
I've increased the cooldown to 8 seconds, which should be acceptable for a test and should cut CI and other potentially-slow environments some more slack.